### PR TITLE
Remove build profiles

### PR DESCRIPTION
**Pull request by blairconrad**

### CODE REVIEW

**Comment by blairconrad** [comment]
I didn't think the test helpers project was pulling its weight anymore, so in an opinionated move, I collapsed it first.

File: how_to_build.md [hunk 1]
```diff
@@ -5,9 +5,7 @@ It also replicates the build on the Continuous Integration build server and is t
 
 You can also build the solution using Visual Studio 2019 or later, but this doesn't provide the same assurances as the command line build.
 
-At the time of writing the full build (including all target frameworks) can only run on Windows.
-
-[Partial builds](#building-only-a-subset-of-the-supported-target-frameworks) are supported on Linux (Ubuntu 18.04). This might also run on macOS, but hasn't been tested.
+At the time of writing the full build (including testing on all target frameworks) can only run on Windows.
```
Comment (thomaslevesque) [inline how_to_build.md]: I thought we'd test only on .NET Core, actually. But I guess it doesn't hurt to keep testing on .NET Framework as well.

File: how_to_build.md [hunk 2]
```diff
@@ -29,11 +27,9 @@ Ensure that the following are installed:
 
 6. A recent version of the .NET Core 3.0 SDK (currently this means 3.0.100 or later)
 
-You might not need everything to run a [partial build](#building-only-a-subset-of-the-supported-target-frameworks).
-
 ### On Linux
 
-The default [build profile](#building-only-a-subset-of-the-supported-target-frameworks) on Linux is `netcore2+`, which runs the tests on .NET Core 2.1 and 3.0. The .NET Framework isn't supported on Linux, and .NET Core 1.0 is no longer supported, which makes it difficult (if not impossible) to install on Linux.
+On non-Windows operating systems, tests are run on .NET Core 2.1 and 3.0. The .NET Framework isn't supported on Linux, and .NET Core 1.0 is no longer supported, which makes it difficult (if not impossible) to install on Linux.
```
Comment (thomaslevesque) [inline how_to_build.md]: We're not targeting .NET Core 1.0, so maybe it's not worth mentioning.

File: tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs [hunk 3]
```diff
@@ -27,8 +27,8 @@ public override void After(MethodInfo methodUnderTest)
             Thread.CurrentThread.CurrentCulture = this.originalCulture;
             Thread.CurrentThread.CurrentUICulture = this.originalUiCulture;
 #else
-            CultureInfo.CurrentCulture = this.originalCulture;
-            CultureInfo.CurrentUICulture = this.originalUiCulture;
+            CultureInfo.CurrentCulture = this.originalCulture!;
+            CultureInfo.CurrentUICulture = this.originalUiCulture!;
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: We don't have warnings here in the FakeItEasy solution... Any idea why?

Comment (thomaslevesque) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: Anyway, maybe a better approach would be to initialize `originalCulture` and `originalUICulture` with the current culture in the constructor?

File: how_to_build.md (hunk 1 above)
Comment (blairconrad) [reply how_to_build.md]: I'd be happy to reduce (or keep), but as I think you've mentioned elsewhere, who the heck knows where these analyzers will run, and under what runtime. Like the surprise we got when they stopped working in VS2017 due to the netstandard2.0 update.

Comment (blairconrad) [reply how_to_build.md]: Anyhow, I see no harm in keeping for now.

File: tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs (hunk 3 above)
Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: I do not know why there's no warning in FIE.. I was surprised to see it show up. I can take a look over there sometime.

Happy to initialize in the constructor, if you think it's safe. I just didn't know if the `Before`/`After` are run in a different environment than the attribute code. Do we know if the culture will be the same at constructor time as when the test is run?

Comment (thomaslevesque) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: I think it's safe. Anyway, we'll overwrite it in `Before`, so it probably doesn't matter.

Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: This is a good point.

Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: I pushed a thing, but I don't like it. Hold on.

Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: Okay. I like the new thing a little more than the old thing. Carry on.

File: tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs [hunk 4]
```diff
@@ -11,40 +11,42 @@ namespace FakeItEasy.Analyzer.Tests.Helpers
...
+        private static Action<CultureInfo> setCurrentCulture = culture => Thread.CurrentThread.CurrentCulture = culture;
+        private static Action<CultureInfo> setCurrentUiCulture = culture => Thread.CurrentThread.CurrentUICulture = culture;
+        private static Func<CultureInfo> getCurrentCulture = () => Thread.CurrentThread.CurrentCulture;
+        private static Func<CultureInfo> getCurrentUiCulture = () => Thread.CurrentThread.CurrentUICulture;
+#else
+        private static Action<CultureInfo> setCurrentCulture = culture => CultureInfo.CurrentCulture = culture;
+        private static Action<CultureInfo> setCurrentUiCulture = culture => CultureInfo.CurrentUICulture = culture;
+        private static Func<CultureInfo> getCurrentCulture = () => CultureInfo.CurrentCulture;
+        private static Func<CultureInfo> getCurrentUiCulture = () => CultureInfo.CurrentUICulture;
+#endif
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: Sorry to nitpick, but why delegates rather than methods?

Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: No reason. I can method them.

Comment (thomaslevesque) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: I think it would make the whole thing more readable

Comment (blairconrad) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs]: Methoded.

**Comment by blairconrad** [comment]
Thanks for the careful review and subsequent merge. Sorry for the churn!

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy.Analyzers 6.0.0-beta.1](https://github.com/FakeItEasy/FakeItEasy.Analyzers/releases/tag/6.0.0-beta.1).

**Pull request merged**